integrasjonspunkt/src/main/java/no/difi/meldingsutveksling/mail/EduMailSender.java

```diff
@@ -44,21 +52,33 @@ protected PasswordAuthentication getPasswordAuthentication() {
             message.setSubject(title);
 
             MimeMultipart mimeMultipart = new MimeMultipart();
-            // Add content
             MimeBodyPart mimeBodyPart = new MimeBodyPart();
-            mimeBodyPart.setText("Melding vedlagt.");
-            mimeMultipart.addBodyPart(mimeBodyPart);
-            // Add attachement
-            ByteArrayDataSource ds = new ByteArrayDataSource(messageAsBytes, "application/octet-stream");
-            MimeBodyPart attachementPart = new MimeBodyPart();
-            attachementPart.setDataHandler(new DataHandler(ds));
-            attachementPart.setFileName(title+".xml");
-            mimeMultipart.addBodyPart(attachementPart);
 
+            if (PayloadUtil.isAppReceipt(request)) {
+                mimeBodyPart.setText("Kvittering (AppReceipt) mottatt.");
+            } else {
+                mimeBodyPart.setText("Du har fått en BestEdu melding. Se vedlegg for metadata og dokumenter.");
+
+                Object payload = unmarshallPayload(request.getPayload());
+                List<DokumentType> docs = ((MeldingType) payload).getJournpost().getDokument();
+                docs.forEach(d -> {
+                    ByteArrayDataSource ds = new ByteArrayDataSource(d.getFil().getBase64(), d.getVeMimeType());
+                    MimeBodyPart attachementPart = new MimeBodyPart();
+                    try {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Extract this nested try block into a separate method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://sonar-app01.difi.local:9000/coding_rules#rule_key=squid%3AS1141)